### PR TITLE
#349 Remove privileged containers requirement

### PR DESCRIPTION
**Pull request by ianpittwood**
- Removed `--privileged` flag for all documentation and definitions/scripts except for Connect.
- Added NEWS entry for change with link to Privileged Requirements support article.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -26,7 +25,6 @@ services:
       args:
         RSC_VERSION: 2022.06.2
     image: rstudio/rstudio-connect:2022.06.2
-    privileged: true
```
Comment (colearendt) [inline docker-compose.yml:29]: I think this one is still required?

File: connect-content-init/docker-compose.test.yml [hunk 2]
```diff
@@ -5,7 +5,6 @@ services:
     image: $IMAGE_NAME
     command: /run_tests.sh
     entrypoint: []
-    privileged: true
```
Comment (colearendt) [inline connect-content-init/docker-compose.test.yml:8]: Good catch!

File: package-manager/README.md [hunk 3]
```diff
@@ -109,14 +108,14 @@ to explore using a license server, license file, or custom image with manual int
 export RSPM_LICENSE=XXXX-XXXX-XXXX-XXXX-XXXX-XXXX-XXXX
 
 # Run without persistent data and using an external configuration
-docker run -it --privileged \
```
Comment (colearendt) [inline package-manager/README.md:112]: These are about to be removed by the RSPM team landing a bunch of changes in #348 - probably best to leave off changes to the package-manager side, TBH

File: workbench/README.md [hunk 4]
```diff
@@ -29,7 +29,7 @@ To verify basic functionality as a first step:
 export RSW_LICENSE=XXXX-XXXX-XXXX-XXXX-XXXX-XXXX-XXXX
 
 # Run without persistent data using default configuration
-docker run --privileged -it \
```
Comment (colearendt) [inline workbench/README.md:32]: Just worth noting that workbench needs the other config changes to work in this setup

File: helper/float/README.md [hunk 5]
```diff
@@ -35,7 +35,7 @@ export RSC_LICENSE_SERVER=rsc-float-lic:8999
 export RSPM_LICENSE_SERVER=rspm-float-lic:8969
 
 # RStudio Workbench
-docker run --privileged -it \
```
Comment (colearendt) [inline helper/float/README.md:38]: Nice! Would be good to test these to be sure they never needed `--privileged`

**Review by colearendt — COMMENTED** [review]
Draft is looking great! Just a few points to keep in mind

File: docker-compose.yml (hunk 1 above)
Comment (ianpittwood) [reply docker-compose.yml]: Yes, thank you!

File: workbench/README.md (hunk 4 above)
Comment (ianpittwood) [reply workbench/README.md:32]: I added launcher.local.conf, are there additional changes that are needed?

File: package-manager/README.md (hunk 3 above)
Comment (ianpittwood) [reply package-manager/README.md]: I rebased after their PR merged so I think we should be good now, right?

File: helper/float/README.md (hunk 5 above)
Comment (ianpittwood) [reply helper/float/README.md:38]: I tested the float containers and they seem to be working OK.

**Review by tylfin — APPROVED** [review]

**Review by colearendt — COMMENTED** [review]
LGTM! Only nit is that a NEWS.md entry in workbench probably makes sense since we changed the default configuration for the container

**Comment by ianpittwood** [comment]
@colearendt I added an entry in workbench/NEWS.md per your feedback. Let me know if you'd like different wording!

**Review by colearendt — APPROVED** [review]
LGTM!

**Pull request merged**